enterprise/frontend/src/embedding-sdk/components/private/util/FlexibleSizeComponent.tsx

```diff
@@ -0,0 +1,31 @@
+import type { CSSProperties, PropsWithChildren } from "react";
+
+import { Box } from "metabase/ui";
+
+export type FlexibleSizeProps = PropsWithChildren<{
+  height?: CSSProperties["height"];
+  width?: CSSProperties["width"];
+  className?: string;
+  style?: CSSProperties;
+}>;
+
+export const FlexibleSizeComponent = ({
+  height: propHeight = "600px",
+  width: propWidth = "100%",
+  className,
+  style,
+  children,
+}: FlexibleSizeProps) => (
+  <Box h={propHeight} w={propWidth} mih={propHeight} miw={propWidth}>
+    <Box
+      h="100%"
+      w="100%"
+      mih="100%"
+      miw="100%"
```

Comment:
I wonder if `min-height: 100%` and `min-width: 100%` does really affect the style?

Reply:
I think it's a good way to ensure that the component doesn't shrink unexpectedly like it did before when the chart selectors were closed. I remember when we tried this without the min-width and height on the `QuestionVisualization` component and we still had issues with sizing. This should solve it 🤞